### PR TITLE
Import ed25519

### PR DESCRIPTION
**Pull request by 95DDB**

### CODE REVIEW

File: algebra/src/curves/ed25519/tests.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    #[test]
+    fn test_projective_group() {
+        let a = rand::random();
+        let b = rand::random();
+        for _i in 0..100 {
+            group_test::<TEEd25519Projective>(a, b);
+        }
+    }
```
Comment (UlrichHaboeck75) [inline algebra/src/curves/ed25519/tests.rs:39]: why running 100 times group_test, if curve_tests() does already extensive arithmetic checks?

File: algebra/src/curves/ed25519/tests.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    #[test]
+    fn test_affine_group() {
+        let a: TEEd25519Affine = rand::random();
+        let b: TEEd25519Affine = rand::random();
+        for _i in 0..100 {
+            group_test::<TEEd25519Affine>(a, b);
+        }
```
Comment (UlrichHaboeck75) [inline algebra/src/curves/ed25519/tests.rs:47]: Is this a workaround as curve_tests() is templatized by ProjectiveCurve?

File: algebra/src/curves/ed25519/tests.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        for _i in 0..100 {
+            group_test::<SWEd25519Projective>(a, b);
+        }
+    }
+
+    #[test]
+    fn test_generator() {
+        let generator = SWEd25519Affine::prime_subgroup_generator();
+        assert!(generator.is_valid());
+    }
```
Comment (UlrichHaboeck75) [inline algebra/src/curves/ed25519/tests.rs:93]: Why we have this test although we have the sage script for checking curve parameters?

File: algebra/src/fields/ed25519/tests.rs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    field_test(a, b);
+    primefield_test::<Fr>();
+}
+
+#[test]
+fn test_ed25519_fq() {
+    let a: Fq = rand::random();
+    let b: Fq = rand::random();
+    field_test(a, b);
+    primefield_test::<Fq>();
```
Comment (UlrichHaboeck75) [inline algebra/src/fields/ed25519/tests.rs:17]: Why we do not do the sqrt_field_test?

**Review by UlrichHaboeck75 — CHANGES_REQUESTED** [review]
I re-run our test scripts, the Edwards parameters are fine. I posted some minor questions about the tests. When these are answered we can merge anytime.

File: algebra/src/fields/all_field_checks.sh [hunk 5]
```diff
@@ -0,0 +1,86 @@
+echo "################ CHECKING ALL FIELD PARAMETERS ##################"
```
Comment (UlrichHaboeck75) [inline algebra/src/fields/all_field_checks.sh:1]: @DDT92 Not related to this file, but the check_field_parameters.sage: 
Please optimize the test for the root of unity.

File: algebra/src/curves/ed25519/tests.rs (hunk 2 above)
Comment (95DDB) [reply algebra/src/curves/ed25519/tests.rs:47]: Yes, you answered yourself

File: algebra/src/curves/ed25519/tests.rs (hunk 3 above)
Comment (95DDB) [reply algebra/src/curves/ed25519/tests.rs:93]: It's just a consistency test. I would keep it.

File: algebra/src/curves/ed25519/tests.rs (hunk 1 above)
Comment (95DDB) [reply algebra/src/curves/ed25519/tests.rs:39]: See below

File: algebra/src/fields/ed25519/tests.rs (hunk 4 above)
Comment (95DDB) [reply algebra/src/fields/ed25519/tests.rs:17]: Done

File: algebra/src/fields/all_field_checks.sh [hunk 6]
```diff
@@ -37,12 +37,6 @@ sage check_field_parameters.sage bls12_377/fr.rs #
 echo "############ Scalar field:"
 sage check_field_parameters.sage edwards_sw6/fr.rs #
 echo #
-echo "###############Checking edwards_sw6 field parameters:"
-echo "############ Base field:"
-sage check_field_parameters.sage sw6/fr.rs #
-echo "############ Scalar field:"
-sage check_field_parameters.sage edwards_sw6/fr.rs #
-echo #
```
Comment (UlrichHaboeck75) [inline algebra/src/fields/all_field_checks.sh]: @DDT92 why are these fields removed from the test?

File: algebra/src/fields/all_field_checks.sh (hunk 5 above)
Comment (DDT92) [reply algebra/src/fields/all_field_checks.sh:1]: Done.

**Review by DDT92 — APPROVED** [review]

**Review by albertog78 — APPROVED** [review]

**Review by UlrichHaboeck75 — APPROVED** [review]

**Pull request merged**